### PR TITLE
HC Dashboard - Child Wellness, Nutrition page

### PR DESCRIPTION
**Pull request by anvmn**
#982

### CODE REVIEW

**Comment by anvmn** [comment]
Selection_103

**Comment by anvmn** [comment]
Selection_106

**Comment by anvmn** [comment]
Selection_108

**Comment by anvmn** [comment]
Peek 2024-01-04 11-50

File: client/src/elm/Pages/WellChild/Activity/Update.elm [hunk 1]
```diff
@@ -440,26 +440,37 @@ update currentDate site isChw id db msg model =
...
 
                             else if zscore < -3 then
-                                WarningHeadCircumferenceMicrocephaly
+                                ( WarningHeadCircumferenceMicrocephaly
+                                , [ setEncounterWarningMsg WarningHeadCircumferenceMicrocephaly ]
+                                )
 
                             else
-                                NoHeadCircumferenceWarning
+                                -- Z-score value is within range, so not warning is
```
Comment (AronNovak) [inline client/src/elm/Pages/WellChild/Activity/Update.elm]:
```suggestion
                                -- Z-score value is within range, so no warning is
```

File: server/hedley/modules/custom/hedley_admin/scripts/assign-nutrition-encounter-type.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ * profiles/hedley/modules/custom/hedley_admin/scripts/assign-nutrition-encounter-type.php.
+ */
+
+if (!drupal_is_cli()) {
+  // Prevent execution from browser.
+  return;
+}
+
+// Get the number of nodes to be processed.
+$batch = drush_get_option('batch', 500);
```
Comment (AronNovak) [inline server/hedley/modules/custom/hedley_admin/scripts/assign-nutrition-encounter-type.php:20]: Is this batch size safe enough? Tested with live DB?

File: server/hedley/modules/custom/hedley_admin/scripts/fix-well-child-encounters-warnings.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+// Get the number of nodes to be processed.
+$batch = drush_get_option('batch', 500);
+
+// Get allowed memory limit.
+$memory_limit = drush_get_option('memory_limit', 200);
+
+// Locate all Head Circumferences measurements that got value.
+// Note: when skipped, value is ste to 0.
```
Comment (AronNovak) [inline server/hedley/modules/custom/hedley_admin/scripts/fix-well-child-encounters-warnings.php]:
```suggestion
// Note: when skipped, value is set to 0.
```

File: server/hedley/modules/custom/hedley_stats/hedley_stats.module [hunk 4]
```diff
@@ -1557,6 +1798,337 @@ function hedley_stats_add_immunization_dates_for_bundle(SelectQuery $base_query,
...
+
+  $result = hedley_stats_run_node_query_in_batches($query);
+
+  $items = [];
+  // Prepare participant data to be consumed by client side.
+  foreach ($result as $row) {
+    $items[$row->nid] = [
+      'id' => $row->field_person,
+      // Field for 'created' is added at hedley_stats_get_base_query().
+      'created' => date_format(date_create($row->field_expected), 'Y-m-d'),
```
Comment (AronNovak) [inline server/hedley/modules/custom/hedley_stats/hedley_stats.module:1867]: maybe put date format in a constant, it seems to be redundant

File: server/hedley/modules/custom/hedley_stats/hedley_stats.module [hunk 5]
```diff
@@ -1557,6 +1798,337 @@ function hedley_stats_add_immunization_dates_for_bundle(SelectQuery $base_query,
...
+  // Join with Nutrition Encounter table, through which we can
+  // join measurements data.
+  $base_query->leftJoin('field_data_field_nutrition_encounter', 'nutrition_encounter', 'nutrition_encounter.field_nutrition_encounter_target_id = node.nid');
+
+  $query = clone $base_query;
+
+  // Pull data of MUAC that was recorded during encounter.
+  $query->leftJoin('field_data_field_muac', 'muac', 'muac.entity_id = nutrition_encounter.entity_id');
+  $query->addField('muac', 'field_muac_value');
+  $query->addExpression("GROUP_CONCAT(DISTINCT muac.field_muac_value)", 'muac');
```
Comment (AronNovak) [inline server/hedley/modules/custom/hedley_stats/hedley_stats.module:1896]: Don't we need all MUAC values here? With DISTINCT, we may drop some of the values.

File: server/hedley/modules/custom/hedley_admin/scripts/assign-nutrition-encounter-type.php (hunk 2 above)
Comment (anvmn) [reply server/hedley/modules/custom/hedley_admin/scripts/assign-nutrition-encounter-type.php:20]: yep

File: server/hedley/modules/custom/hedley_stats/hedley_stats.module (hunk 5 above)
Comment (anvmn) [reply server/hedley/modules/custom/hedley_stats/hedley_stats.module:1896]: Well, actually, here we add data of MUAC measurement to individual encounter measurements. There can be only single MUAC measurement, so we should be safe with DISTINCT.

File: server/hedley/modules/custom/hedley_stats/hedley_stats.module (hunk 4 above)
Comment (anvmn) [reply server/hedley/modules/custom/hedley_stats/hedley_stats.module:1867]: Well, `Y-m-d` is standard format we use in project.
I don't really see a reason for putting it at constant, unless at a time when we will be required to change the format.
This is not likely to happen, but if we'll need to do it, I'll make it a constant then.

**Review by AronNovak — APPROVED** [review]

**Pull request merged**